packages/evm/test/adapters/Sygma/03_SygmaMessageRelayer.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { expect } from "chai"
+import { ethers, network } from "hardhat"
+
+const DOMAIN_ID = 5
+const resourceID = "0x0000000000000000000000000000000000000000000000000000000000000500"
+
+const setup = async () => {
+  await network.provider.request({ method: "hardhat_reset", params: [] })
+  const signers = await ethers.getSigners()
+  const sender = signers[0].address
+  const adapter = signers[1].address
+  const otherAddress = signers[2].address
+  const Yaho = await ethers.getContractFactory("Yaho")
+  const yaho = await Yaho.deploy()
+  const SygmaBridge = await ethers.getContractFactory("MockSygmaBridge")
+  const sygmaBridge = await SygmaBridge.deploy()
+  const SygmaMessageRelayer = await ethers.getContractFactory("SygmaMessageRelayer")
+  // IBridge bridge, HeaderStorage headerStorage, bytes32 resourceID, uint8 defaultDestinationDomainID, address defaultSygmaAdapter
+  const sygmaMessageRelayer = await SygmaMessageRelayer.deploy(
+    sygmaBridge.address,
+    yaho.address,
+    resourceID,
+    DOMAIN_ID,
+    adapter,
+    "0x00",
+  )
+  const PingPong = await ethers.getContractFactory("PingPong")
+  const pingPong = await PingPong.deploy()
+  const message_1 = {
+    to: pingPong.address,
+    toChainId: 1,
+    data: pingPong.interface.getSighash("ping"),
+  }
+  await yaho.dispatchMessages([message_1, message_1])
+  // await mine(10)
+  return {
+    sender,
+    adapter,
+    otherAddress,
+    yaho,
+    sygmaBridge,
+    sygmaMessageRelayer,
+    pingPong,
+    message_1,
+  }
+}
+
+const prepareDepositData = async (reporterAddress: string, ids: string[], hashes: string[], adapter: string) => {
+  const abiCoder = ethers.utils.defaultAbiCoder
+  const executionData = abiCoder
+    .encode(["address", "uint256[]", "bytes32[]"], [ethers.constants.AddressZero, ids, hashes])
+    .substring(66)
+
+  const SygmaAdapter = await ethers.getContractFactory("SygmaAdapter")
```

Comment:
Did you mean?
```js
  const PingPong = await ethers.getContractFactory("PingPong")
  const functionSig = PingPong.interface.getSighash("ping")
  ```

Reply:
No, I already got the functionSig for ping, here I meant to get the functionSig for storeHashes in SygmaAdapter on the target side